### DISCUSSION TITLE
Identification Failures near Glass Windows and Glasses

### DISCUSSION THREAD

**Discussion by MateusValasquesIOB** [vladmandic/human#449] [Q&A]
**Issue Description**
We are having some issues regarding reflections in glass windows and glasses, do you have some suggestions in how we could approach this problem and improve identification?

Even though the person was registered with the incorporation, without reflections and without glasses, the lib still incorrectly identified the person;

**Steps to Reproduce**
When trying to identify someone close to a glass window or that is using glasses, the identification fails, mismatching the person with another person;
**Expected Behavior**
It is expected that glasses interfere with the process, but it was unexpected that glass windows wouldn't.
**Environment**

- Human library version? 3.2.1
- Built-in demo or custom code? custom code
- Type of module used (e.g. `js`, `esm`, `esm-nobundle`)? esm
- TensorFlow/JS version (if not using bundled module)?
- Browser or NodeJS and version (e.g. *NodeJS 14.15* or *Chrome 89*)? Chrome 123
- OS and Hardware platform (e.g. *Windows 10*, *Ubuntu Linux on x64*, *Android 10*)? Windows 10
- Packager (if any) (e.g, *webpack*, *rollup*, *parcel*, *esbuild*, etc.)?
- Framework (if any) (e.g. *React*, *NextJS*, etc.)? Angular

**Diagnostics**

- Check out any applicable [diagnostic steps](https://github.com/vladmandic/human/wiki/Diag)

**Additional**
- Human Config: 
- humanConfig = {
    debug: false,
    modelBasePath: '../assets/models/',
    filter: { enabled: true, equalization: true, flip: false, width: 0 },
    face: {
      enabled: true,
      detector: { rotation: true, maxDetected: 100, minConfidence: 0.6, return: true, mask: false },
      iris: { enabled: true },
      description: { enabled: true }, // default model for face descriptor extraction is faceres

      mesh: { enabled: true },
      emotion: { enabled: false },
      antispoof: { enabled: true },
      liveness: { enabled: true },
    },
    body: { enabled: false },
    hand: { enabled: false },
    object: { enabled: false },
    gesture: { enabled: true },
    segmentation: { enabled: false },
  };
- For installation or startup issues include your `package.json`
- For usage issues, it is recommended to post your code as [gist](https://gist.github.com/)
- For general questions, create a [discussion topic](https://github.com/vladmandic/human/discussions)

**Comment by vladmandic** [comment]
what do you mean by "reflection in glass window"? if you see a full reflection, then its detected as a different person than primary and you should set maximum number of detected people - for example, 1.
regarding glasses, its a known problem - and honestly i didn't put too much time into solving it since entire concept of person identification is a value add here, its not a primary functionality..

**Comment by vladmandic** [comment]
moving this to discussions as its not really a product issue as such.
